[LE12] linux update to 6.6.57

sky42src opened the pull request:
build tested with my community release 12.0.1-#241027 for Generic, RPi, AW, RK
run tested on Generic, RPi4, RPi5

RK needed some patch adjustments  
deleted:  projects/Rockchip/patches/linux/default/linux-0003-pinctrl--rockchip--correct-RK3328-iomux-width-flag-for-GPIO2-B-pins.patch
modified: projects/Rockchip/patches/linux/default/linux-1000-drm-rockchip.patch removed hunks 11 and 47

chewitt commented:
@sky42src can you please confirm the dropped patches/hunks are the result of upstream merges? (not simply things that aren't applying anymore).

sky42src commented:
I am not really sure how to look that up, but patch asked if it should reverse the patch.  
For all the hunks i removed i did it only when `scripts/unpack linux` stated  
`Reversed (or previously applied) patch detected!  Assume -R? [n]` and only then i removed this hunk.  
For the big patch i split it up and checked hunk by hunk.

vpeter4 commented:
> I am not really sure how to look that up

Open the patch file and linux source file and compare the content.

sky42src commented:
Sorry but me reading source code and comparing it manually to the patch is definitively less reliable then patch saying `Reversed (or previously applied) patch detected!  Assume -R? [n]`.  
I checked that hunk by hunk with `scripts/unpack linux` and for that i split up the big patch of 2972 lines in one file per hunk. If that is not good enough just close the PR and i carry that my self in my community release.

vpeter4 commented:
Checking source code just adds some more confirmation.

And btw: this PR is ok :)

heitbaum commented on projects/Rockchip/patches/linux/default/linux-0003-pinctrl--rockchip--correct-RK3328-iomux-width-flag-for-GPIO2-B-pins.patch:
Part of 6.6.49
https://lwn.net/Articles/988749/

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/pinctrl/pinctrl-rockchip.c?h=v6.6.57#n3805

heitbaum commented on projects/Rockchip/patches/linux/default/linux-1000-drm-rockchip.patch:
Part of 6.6.54
https://lwn.net/Articles/992980/

      drm/rockchip: dw_hdmi: Fix reading EDID when using a forced mode
      clk: rockchip: Set parent rate for DCLK_VOP clock on RK3228

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/clk/rockchip/clk-rk3228.c?h=v6.6.57#n412
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/gpu/drm/rockchip/rockchip_drm_vop.c?h=v6.6.57#n385
